src/main/java/org/opendevstack/provision/services/BitbucketAdapter.java

```diff
@@ -651,8 +651,9 @@ protected BitbucketProjectData callCreateProjectApi(OpenProjectData project) thr
     return projectData;
   }
 
-  protected RepositoryData callCreateRepoApi(String projectKey, Repository repo)
+  protected RepositoryData callCreateRepoApi(OpenProjectData project, Repository repo)
```

Comment:
A small improvement, but I think it improves readability from the method caller point of view. I would prefer here to  additionally pass the boolean `isSpecialPermissionSet` instead of passing the object `OpenProjectData`. The method signature would look like this:
``` 
protected RepositoryData callCreateRepoApi(String projectKey, boolean specialPermissionSet, Repository repo)
``` 
Can you please change it?

Reply:
Well, I thought that passing an independent flag could be error prone or misinterpreted to allow customization. It also feels like showing out the implementation. If, in the future, we changed the criteria to assign the permissions, the flag could become useless and additional parameters may be needed.
That's why I  decided to pass the project information as a whole, even if we are currently only using the key and that flag.
What do you think?

Reply:
If we need to change the criteria the in the future, the code will be changed anyway. In this case having a method signature that explicitly defines what it uses makes the code easier to read. I think, readability wins here.

Reply:
Ok, done